### PR TITLE
Bugfix - indent and linemove

### PR DESCRIPTION
**Pull request by harshad1**
This PR fixes 2 bugs:

1. Indent / de-indent did not work correctly when the cursor was inside the leading whitespace
2. Line move up or down would incorrectly trigger auto formatting.

In addition, the line move functionality has been updated to preserve selection (can use linemove multiple times now) and supports more than 1 selected line.

### CODE REVIEW

**Comment by gsantner** [comment]
still wip?

**Comment by harshad1** [comment]
Sorry, had a busy weekend, this is good to go.

File: app/src/main/java/net/gsantner/opoc/util/StringUtils.java [hunk 1]
```diff
@@ -130,16 +142,21 @@ public static int getIndexFromLineOffset(final CharSequence s, final int[] le) {
...
+                    }
+                    if (count == l) {
+                        break;
+                    }
                 }
             }
-            if (i < s.length()) {
-                return getLineEnd(s, i + 1) - e;
+            if (i < s.length() - 1) {
+                final int start = (l == 0) ? 0 : i + 1;
```
Comment (gsantner) [inline app/src/main/java/net/gsantner/opoc/util/StringUtils.java:156]: `1 == 0` is always fale

File: app/src/main/java/net/gsantner/markor/format/general/CommonTextActions.java [hunk 2]
```diff
@@ -223,76 +230,51 @@ public void onClick(DialogInterface dialog, int selectedColor, Integer[] allColo
...
-        String lineToMove = _hlEditor.getText().toString().substring(_hlEditor.getSelectionStart(), _hlEditor.getSelectionEnd());
-        if (!lineToMove.endsWith(LINE_SEPARATOR)) {
-            lineToMove += LINE_SEPARATOR;
-        }
-        _hlEditor.simulateKeyPress(KeyEvent.KEYCODE_DEL);
-        _hlEditor.simulateKeyPress(up ? KeyEvent.KEYCODE_DPAD_UP : KeyEvent.KEYCODE_DPAD_DOWN);
-        _hlEditor.simulateKeyPress(KeyEvent.KEYCODE_MOVE_HOME);
-        _hlEditor.getText().insert(_hlEditor.getSelectionStart(), lineToMove);
-    }
+                final String newPair = String.format("%s\n%s", up ? line : altLine, up ? altLine : line);
```
Comment (gsantner) [inline app/src/main/java/net/gsantner/markor/format/general/CommonTextActions.java:265]: just seeing concat of two lines hair. sure no issue with multiple lines selected?

**Comment by harshad1** [comment]
I'm sure. I should probably change the variable name from line to 'lines' though

H
Aug. 18, 2020 13:01:19 Gregor Santner <notifications@github.com>:

**Comment by harshad1** [comment]
this is good to go

**Comment by gsantner** [comment]
thx merged

**Pull request merged**